upt_macports/upt_macports.py

```diff
@@ -207,6 +228,14 @@ def _normalized_macports_folder(name):
     def jinja2_reqformat(self, req):
         return f'rb${{ruby.suffix}}-{req.name.lower()}'
 
+    @property
+    def homepage(self):
+        homepage = self.upt_pkg.homepage
+        if homepage.startswith('http'):
+            return homepage
+        else:
+            return f'https://rubygems.org/gems/{self.upt_pkg.name}'
+
```

Comment:
So, in the case of Ruby, using this default URL is already done in upt-rubygems (see https://framagit.org/upt/upt-rubygems/blob/master/upt_rubygems/upt_rubygems.py#L170). Obviously the maintainer could set the homepage to "http://dead-link.com", but hopefully that does not happen too often :)

Reply:
right, but currently if the metadata contains something random as ```unknown``` or whatever, it will still return that as URL... that's not what we want. It should be set to a default if it's not a valid URL.